### DISCUSSION TITLE
question about selfhosted relay and id servers

### DISCUSSION THREAD

**Discussion by stamepicmorg** [rustdesk/rustdesk#1834] [Q&A] [answered]
Hello. I read docs, but i have a question.

* If i setup selfhosted relay and id servers and confiruge desktop (or mobile) clients for them - could i connect to another clients who not connected to my servers?
* Will other clients be able to connect to me if they don't have my server added in settings?

@derStephan also have same [question](https://github.com/rustdesk/rustdesk/discussions/1026#discussioncomment-3431063).

Thanks and best regards.

**Comment by pedjas** [comment, marked as answer]
Both parties have to be connected to the same host.

Other limitation is that both parties may run only one instance of RustDesk at the time. You cannot have one RustDesk client connected to one host and the other connected to other host at the same time. If You run several instances, only the first one started would actually work.
